aiohttp/web_reqrep.py

```diff
@@ -381,8 +381,17 @@ def post(self):
         self._post = MultiDictProxy(out)
         return self._post
 
-    def copy(self):
-        raise NotImplementedError
+    def copy(self, *, app=None, message=None, payload=None, transport=None,
+             reader=None, writer=None, secure_proxy_ssl_header=None):
+        return Request(
```

Comment:
`Request` is a dict-like object and other middlewares can modify it
so all data from original one must be moved to its clone.